### ISSUE TITLE
OM4 boundary forcing is not working

### ISSUE THREAD

**Issue by adele157** [COSIMA/mom6-panan#27]
I've been trying to run a panan-01-zstar case with OM4 boundary forcing. It's been running fine, but something is clearly wrong! 

<img width="866" alt="Screen Shot 2023-02-27 at 1 43 02 pm" src="https://user-images.githubusercontent.com/8506963/221461046-8df6bc4a-7e74-48c6-9592-4f20042e0d9e.png">

<img width="869" alt="Screen Shot 2023-02-27 at 1 43 28 pm" src="https://user-images.githubusercontent.com/8506963/221461064-e917caf0-3349-44bc-8958-94c1011e36b7.png">

I'm not going to have a chance to look into this more until the end of the week. Any chance someone could try to figure out what the problem is? @schmidt-christina @angus-g 

Run directory is here: 
/home/157/akm157/mom6/panan-01-zstar/

Output is here:
/g/data/ik11/outputs/mom6-panan/panan-01-zstar-om4BC/

**Comment by adele157** [comment]
Also the script I used to make the boundary forcing is here:
/g/data/v45/akm157/jupyter_scripts/panan_diagnostics/OM4_panan_boundary_forcing_01.ipynb

And boundary forcing file is here:
/g/data/v45/akm157/inputs/mom6/panan/panan_01/forcing_obc_om4_01.nc

**Comment by adele157** [comment]
I think I figured it out. Looks like my om4 forcing is not on the supergrid (compare /g/data/v45/akm157/inputs/mom6/panan/panan_01/forcing_obc_om4_01.nc to /g/data/ik11/inputs/mom6/panan-01/forcing/forcing_obc_shifted.nc).

Mine has 7200 x points, but perhaps it is supposed to have 14400 x points. Though I'm a bit confused, because I thought 7200 points should be the super grid for the 1/10th, and that matches the number of x points in /g/data/ik11/inputs/mom6/panan-01/ocean_hgrid.nc

@angus-g any thoughts?

**Comment by adele157** [comment]
I noticed a slight difference in the values of the coordinate `nx_segment_001`. Mine increases in steps of 1, but `/g/data/ik11/inputs/mom6/panan-01/forcing/forcing_obc.nc` increases in steps of 2. Do you think that might be the problem? Easy to try to see if it makes a difference at least.

**Comment by PaulSpence** [comment]
Are you sure there's no Nan's :)

**Comment by adele157** [comment]
Yep, no NaNs. But I am totally clutching at straws, so any more half-brained ideas are very welcome.

**Comment by adele157** [comment]
Changing nx_segment_001 to increase by steps of 2 did not help.

**Comment by adele157** [comment]
Also discovered the time variable was a bit weird. Doing a couple more tests now.

**Comment by adele157** [comment]
Update: I think the problem may be coming from the new initial conditions not the OM4 boundary forcing. @fabiobdias 
I should have known better than to try and change 2 things at once...

Green and black here use the same original ACCESS-OM2 boundary forcing. Only difference is green uses the new initial conditions.
<img width="611" alt="Screen Shot 2023-03-03 at 6 35 40 am" src="https://user-images.githubusercontent.com/8506963/222534072-25efda5a-c3e1-44a6-8188-1354548b7d4b.png">

If anyone could have a look and see if they can spot any obvious problems in these files that would be really helpful. @angus-g @aekiss 

New initial condition file is here:
/g/data/v45/akm157/inputs/mom6/panan/panan_01/ocean_pottemp_salt.res.nc

Original panan initial condition file is here:
/g/data/ik11/inputs/mom6/panan-01/forcing/temp_salt_init_z.nc

**Comment by adele157** [comment]
Actually, thinking about it a bit more:
Maybe the overturning is just lower in the new initial conditions. I'm running this out a bit more to see if the overturning stays at this value or goes wacky. Will also try the combination of original initial condition file and new boundary forcing.

**Comment by adele157** [comment]
There's definitely a problem with those new initial conditions @fabiobdias. This is the overturning at the northern boundary during year 2. Interestingly the case with the old IC, but new boundary forcing (not shown here) looks similar. I wonder if neither of them are picking up the boundary forcing data.
<img width="611" alt="Screen Shot 2023-03-04 at 8 45 15 am" src="https://user-images.githubusercontent.com/8506963/222835731-4c4188ce-8314-4898-9a17-bcdbe48779b0.png">

**Comment by fabiobdias** [comment]
Hi @adele157 I've looked at the surface temperature and salinity in the new and old IC and they look both ok - the only difference is that the new IC don't have fill values at land points, but I'm not sure that matters. Also, did you say the experiment with old IC and the OM4 boundary forcing looks similar to the red or black line in the figure above?

<img width="727" alt="Screenshot 2023-03-04 at 9 45 18 am" src="https://user-images.githubusercontent.com/8631257/222845965-e56caaac-199c-4df4-a4f4-117f1fce3bcf.png">

**Comment by adele157** [comment]
Old IC and new OM4 BF looks similar to the red above. I'm trying a revised new IC file now (with removed time index and some slight changes to the x/y coords to match the old one).

**Comment by adele157** [comment]
That test of different initial conditions didn't work. The bottom layer has a strong southward transport, instead of northward bottom water transport (this is just raw transports now, not overturning).

Could this weird behaviour be a result of the mismatch between the old boundary forcing (~ yr 200 of ACCESS-OM2-01) and new initial conditions (WOA)? 
<img width="611" alt="Screen Shot 2023-03-04 at 9 04 05 pm" src="https://user-images.githubusercontent.com/8506963/222893866-7b77ed9f-0b6d-4a5b-93c2-aa59576cf7cc.png">

I'm currently running a couple of years of ACCESS-OM2-01 from WOA initial conditions to see if that works better instead of the new OM4 boundary forcing. 

Any help would be very welcome, I think I'm not far off giving up...

**Comment by adele157** [comment]
Update on this:

@angus-g, @fabiobdias and I have been poking around trying to get the ACCESS-OM2-01 yr2 boundary forcing working with not much luck. @fabiobdias and I had a couple of ideas that might be worth trying:
* Swapping in data from the yr 2 output into the old working boundary forcing file (which uses year 156) to make sure we haven't got the coords or variable formatting wrong somehow. 
* Running the script used to create the new yr 2 boundary forcing on the old yr 156 data to see if it is a problem with the actual forcing data. Although I can't find the old data that we used - it used to be here /g/data/ik11/outputs/access-om2-01/01deg_jra55v13_ryf9091/output621{-624}/ but there is no longer 3d daily data there.
* Comparisons of the data in the various forcing files to see if anything jumps out.

Does anyone have suggestions of other things to try?

**Comment by fabiobdias** [comment]
I have plot some of the forcing fields in the northern open boundary (temp/salt/v-vel):

- forcing_obc.nc = original OBC from access-om2-01 year 156
- forcing_access_yr2_01.nc = OBC from access-om2-01 year 2
- forcing_obc_om4_01_fixed-time.nc = OBC from OM4

there are large differences in the temperature and salinity at depth (around layers 30-50), which seems to be caused by deeper thermo/haloclines in access-om2 compared with OM4.

![mom6-panan-01_OBC_temp_comparison](https://user-images.githubusercontent.com/8631257/225159389-22cc3a9d-22b6-4ba3-b01c-880953b78740.png)

![mom6-panan-01_OBC_salt_comparison](https://user-images.githubusercontent.com/8631257/225159408-675a39f9-09b7-40cc-bb80-600c478635f4.png)

![mom6-panan-01_OBC_v_comparison](https://user-images.githubusercontent.com/8631257/225159451-4aa9b0ce-b593-485d-a099-36923a21f406.png)

**Comment by adele157** [comment]
Great, thanks for looking at this @fabiobdias. The OM4 data is on a different vertical grid - so that may be why the anomalies there look so large? The dz_ variables will give the correct depth distribution of the OM4 data.

Also, just noting that I updated you and Angus on slack - I found a problem with some of the attributes on the boundary forcing files. I am rerunning with both the yr 2 OM4 and ACCESS forcing now with these fixed and will update today about whether we're still getting the weird southward abyssal transport at the boundary as in the above plots.

**Comment by adele157** [comment]
Solved!! Turns out the weird southern abyssal transport at the northern boundary was because I increased the tracer time step. Who would have thought!?

This shows the transport in each layer at 37S. The black is the original panan-01. The others are my recreations of that with different tracer timesteps and forcing.

<img width="673" alt="Screen Shot 2023-03-22 at 11 02 29 am" src="https://user-images.githubusercontent.com/8506963/226768900-1961938c-b051-4039-889a-02b7e217e37b.png">

I will redo all the different tests now with the new OM4 and ACCESS-OM2-01 forcing with DT_THERM = 1800.

Perhaps we can continue the discussion of why this went so bad with the higher tracer timestep in https://github.com/COSIMA/mom6-panan/issues/28.
